scripts/custom/clean_dateTime.py

```diff
@@ -0,0 +1,53 @@
+import datetime
+
+from scripts import utils
+
+
+def clean_dateTime(raw_input):  # noqa: C901
+    if not isinstance(raw_input, str):
+        raw_input = str(raw_input)
+    if utils.is_empty(raw_input):
+        return ""
+
+    date = None
+
+    # Handle YYYY
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y")
+        result = date.strftime("%Y")
+    except ValueError:
+        pass
+
+    # Handle YYYYMM
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y%m")
+        result = date.strftime("%Y%m")
+    except ValueError:
+        pass
+
+    # Handle YYYY-MM-DD
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y-%m-%d")
+        result = date.strftime("%Y%m%d")
+    except ValueError:
+        pass
+
+    # Handle YYYY-MM-DDTH:M:S
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y-%m-%dT%H:%M:%S")
+        result = date.strftime("%Y%m%dT%H:%M:%S")
+    except ValueError:
```

Comment:
Here, this format is already fhir friendly; so you don't really need to change it. However, you need to transform something from this format `YYYY-MM-DD H:M:S` to this one `YYYY-MM-DDTH:M:S`.